core/src/main/java/com/capitalone/dashboard/repository/CustomRepositoryQueryImpl.java

```diff
@@ -143,6 +146,29 @@ public CustomRepositoryQueryImpl(MongoTemplate template) {
         return template.find(new Query(c), com.capitalone.dashboard.model.Component.class);
     }
 
+    @Override
+    public Page<CollectorItem> findByCollectorIdInAndJobNameContainingAndNiceNameContainingAllIgnoreCase(List<ObjectId> collectorIds, String jobName, String niceName, Pageable pageable) {
+        Criteria c = Criteria.where("collectorId").in(collectorIds)
+                .and("options.jobName").regex(Pattern.compile(jobName,Pattern.CASE_INSENSITIVE))
+                .and("niceName").regex(Pattern.compile(niceName,Pattern.CASE_INSENSITIVE));
+        Query query = new Query(c);
+        List<CollectorItem> cItems = template.find(query, com.capitalone.dashboard.model.CollectorItem.class);
+        long count = template.count(query,com.capitalone.dashboard.model.CollectorItem.class);
+        Page<CollectorItem> resultPage = new PageImpl<CollectorItem>(cItems,pageable,count);
+        return resultPage;
+    }
+
+    @Override
+    public Page<CollectorItem> findByCollectorIdInAndJobNameContainingIgnoreCase(List<ObjectId> collectorIds, String jobName, Pageable pageable) {
+        Criteria c = Criteria.where("collectorId").in(collectorIds)
+                .and("options.jobName").regex(Pattern.compile(jobName,Pattern.CASE_INSENSITIVE));
+        Query query = new Query(c);
+        List<CollectorItem> cItems = template.find(query, com.capitalone.dashboard.model.CollectorItem.class);
+        long count = template.count(query,com.capitalone.dashboard.model.CollectorItem.class);
+        Page<CollectorItem> resultPage = new PageImpl<CollectorItem>(cItems,pageable,count);
+        return resultPage;
+    }
+
```

Comment:
We dont want these queries in CustomRepositoryQuery class. Please move them to the respective class. I dont think you need a custom query to do REGEX.